src/lib.rs

```diff
@@ -158,7 +225,57 @@ impl<'a> Iterator for SMRecIterator<'a> {
         // } -- End of this stackmap record.
 
         self.num_stackmaps -= 1;
-        Some(Ok(SMRec{id: sm_id, offset: sm_offset}))
+        Some(Ok(SMRec { id, offset, num_locs, locs }))
+    }
+}
+
+struct SMLocIterator<'a> {
+    elf_file: &'a elf::File,
+    cursor: Option<Cursor<&'a Vec<u8>>>,
+    start_pos: u64,
+    num_locs: u16
+}
+
+impl<'a> Iterator for SMLocIterator<'a> {
+    type Item = SMParserResult<SMLoc>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.num_locs == 0 {
+            return None;
+        }
+
+        if self.cursor.is_none() {
+            self.cursor = Some(itry!(cursor_from_elf(self.elf_file, self.start_pos)));
+        }
+
+        let cursor = self.cursor.as_mut().unwrap();
+        // Location[NumRecords] {
+        //     uint8: Register | Direct | Indirect | Constant | ConstIndex
+        let kind = itry!(cursor.read_u8());
+        let kind = itry!(LocKind::from_hex(kind));
+        //     uint8: Reserved (expected to be 0)
+        let reserved = itry!(cursor.read_u8());
+        assert_eq!(reserved, 0);
+        //     uint16: Location Size
+        let size = itry!(cursor.read_u16::<NativeEndian>());
+        //     uint16: Dwarf RegNum
+        let dwarf_reg = itry!(cursor.read_u16::<NativeEndian>());
+        //     uint16: Reserved (expected to be 0)
+        let reserved = itry!(cursor.read_u16::<NativeEndian>());
```

Comment:
Is it intended to overwrite the `reserved` variable before the first one has been used?

Reply:
Yes, both are unused space named `reserved` in the spec. The only reason for the assignment here is to confirm that they are `0` as a sanity check.

Reply:
Is it worth doing the `assert` directly? i.e. something like `assert_eq!(itry!(...), 0)`?

Reply:
agreed

Reply:
Sounds good